Add user names table and some basic admin commands

JesseWilliamson opened the pull request:
This should be fully compatible with the existing database but it's probably a good idea to make a backup copy of timelord.db first anyway. Two new slash commands (/leaderboard and /allusersums) and two new event subscriptions (team_join and user_change) are needed.

JesseWilliamson committed:
Add usernames table

JesseWilliamson committed:
Merge branch 'main' into user-name-log

JesseWilliamson committed:
Merge branch 'main' into user-name-log

JesseWilliamson committed:
Update user table when users change their info

JesseWilliamson committed:
Add functionality for updating user table when users join or update their info

JesseWilliamson committed:
Added admin commands and added support for custom display names

JesseWilliamson committed:
Added comments and general formatting improvements

JesseWilliamson committed:
Merge branch 'main' into user-name-log

ysims commented:
It would be useful to be able to choose a date, and all the data from that date to present date is used. Doesn't need to be in this PR though, can make a new one for that.

JesseWilliamson committed:
Added indentation to make SQL queries more readable

JesseWilliamson committed:
Rearrange and rename functions for consistency / readability

Command and event functions were scattered with no logical order. This groups them into categories and gives them a more readable / consistent layout / naming scheme. This should make it easier to see the progression of what happens when a user submits a command or form.

JesseWilliamson committed:
Removed unnecessary actions

JesseWilliamson committed:
Update descriptions for listener functions and add admin checks

JesseWilliamson committed:
Add a list of commands to readme

JesseWilliamson committed:
Revert "Removed unnecessary actions"

This reverts commit 4248fec92aa75141d5b5056f711d71b7d6294c66.

JesseWilliamson committed:
Gave slack blocks and inputs more consistent names and fixed leaderboard command

LachlanCourt commented on README.md:
```diff
@@ -18,4 +18,18 @@ Add the following environment variables to the `.env` file
+### Slash Commands
+
+#### User Commands:
+- `/timelog` Opens a time logging form
```
I think there's `/help` as well isn't there?

LachlanCourt commented on app.py:
```diff
@@ -18,138 +18,148 @@
+    except Exception as e:
+        # Show the user an error if they input anything other than two integers seperated by some character / characters
+        logger.exception("Invalid user input, failed to create time log entry")
+        respond("*Invalid input!* Please try again!")
```
It might be helpful to provide an example of the expected input as part of this error response

LachlanCourt commented on app.py:
```diff
@@ -172,24 +209,66 @@ def log_database(ack, body, respond, command, logger):
+    ack()
+    try:
+        user_id = body['user_id']
+        num_users = int(command['text']) if command['text'] != "" else 10 # Defaults to 5 entries
```
I think either the comment or the code is incorrect here, is it 5 or 10?

LachlanCourt reviewed (commented):
Just a few notes but otherwise looking good :)

JesseWilliamson committed:
Add /help to commands list and fix leaderboard default

JesseWilliamson committed:
Improve response message timelog error feedback

JesseWilliamson committed:
Merge branch 'main' into user_names

JesseWilliamson committed:
Fixed spelling mistake and removed new time entry format from suggestion

JesseWilliamson committed:
Removed unused database query and fixed display name formatting in /allusersums

LachlanCourt reviewed (approved):
Nice, I'll chuck the new commands up on slack and deploy it on friday :)
